packages/connectivity/src/scp-cf/destination/destination-from-vcap.spec.ts

```diff
@@ -28,34 +34,55 @@ describe('vcap-service-destination', () => {
   });
 
   it('creates a destination using a custom transformation function', () => {
-    const transformationFn = serviceBinding => ({
+    const serviceBindingTransformFn = async (
+      serviceBinding: ServiceBinding
+    ) => ({
       url: serviceBinding.credentials.sys
     });
 
     expect(
-      destinationForServiceBinding('my-custom-service', { transformationFn })
-    ).toEqual({
+      destinationForServiceBinding('my-custom-service', {
+        serviceBindingTransformFn
+      })
+    ).resolves.toEqual({
       url: 'https://custom-service.my.system.com'
     });
   });
 
   it('throws an error if the service type is not supported', () => {
     expect(() =>
       destinationForServiceBinding('my-custom-service')
-    ).toThrowErrorMatchingSnapshot();
+    ).rejects.toThrowErrorMatchingSnapshot();
   });
 
   it('throws an error if no service binding can be found for the given name', () => {
     expect(() =>
       destinationForServiceBinding('non-existent-service')
-    ).toThrowErrorMatchingSnapshot();
+    ).rejects.toThrowErrorMatchingSnapshot();
   });
 
   it('throws an error if there are no service bindings at all', () => {
     delete process.env.VCAP_SERVICES;
     expect(() =>
       destinationForServiceBinding('my-custom-service')
-    ).toThrowErrorMatchingSnapshot();
+    ).rejects.toThrowErrorMatchingSnapshot();
+  });
+
+  it('finds the destination when searching for service bindings', async () => {
+    const serviceBindingTransformFn = async (
+      serviceBinding: ServiceBinding
+    ) => ({
+      url: serviceBinding.credentials.sys
```

Comment:
[pp] Could you perhaps use a obscure specific name like `http://value-from-transform-function.com`. and use in the expect below as well. Then it becomes clear that the method was used. Or use a `spyOn` with `serviceBindingTransformFn` has been called once because we want to ensure that and not that the value is coming from somewhere else, which could happen is you use something from the serviceBinding.cred.sys.

Reply:
Please never use any URLs that are not intended for testing. Rather prefer `example.com/value-from-transform-function` or `value-from-transform-function.example.com` as example.com is owned by IANA explicitly for testing.